sdk/attestation/attestation/src/attestationClient.ts

```diff
@@ -187,18 +187,14 @@ export class AttestationClient {
   ) {
     let credentialScopes: string[] | undefined = undefined;
     let credential: TokenCredential | undefined = undefined;
-    let options: AttestationClientOptions = {};
+    let options: AttestationClientOptions;
 
-    // If arg2 is defined, it's either a tokenCredential or it's a client options.
-    if (credentialsOrOptions !== undefined) {
-      if (isTokenCredential(credentialsOrOptions)) {
-        credential = credentialsOrOptions;
-        credentialScopes = ["https://attest.azure.net/.default"];
-      } else {
-        options = credentialsOrOptions;
-      }
-    } else if (clientOptions !== undefined) {
+    if (credentialsOrOptions && isTokenCredential(credentialsOrOptions)) {
+      credential = credentialsOrOptions;
+      credentialScopes = ["https://attest.azure.net/.default"];
       options = clientOptions;
+    } else {
+      options = credentialsOrOptions || {};
```

Comment:
nit: You can use `??` here. Not a big deal but it is semantically clearer.

Reply:
I don't want to kick off another 100 builds again. 🥲

I'll consider the change if I am making any other significant commit 🙂